xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamFFmpeg.cpp

```diff
@@ -50,18 +50,18 @@ bool CDVDInputStreamFFmpeg::IsEOF()
 
 bool CDVDInputStreamFFmpeg::Open()
 {
-  std::string selected;
   if (m_item.IsInternetStream() && (m_item.IsType(".m3u8") || m_item.GetMimeType() == "application/vnd.apple.mpegurl"))
   {
     // get the available bandwidth and  determine the most appropriate stream
     int bandwidth = CSettings::GetInstance().GetInt(CSettings::SETTING_NETWORK_BANDWIDTH);
     if(bandwidth <= 0)
       bandwidth = INT_MAX;
-    selected = PLAYLIST::CPlayListM3U::GetBestBandwidthStream(m_item.GetPath(), bandwidth);
-    if (selected.compare(m_item.GetPath()) != 0)
+    const CURL playlist_url = m_item.GetURL();
+    const CURL selected = PLAYLIST::CPlayListM3U::GetBestBandwidthStream(playlist_url, bandwidth);
+    if (selected.Get().compare(playlist_url.Get()) != 0)
```

Comment:
This can be changed to `if (selected.Get() != playlist_url.Get())` or, better yet, `if (selected != playlist_url)`. Code clarity and correctness is more important than preserving old code. Don't assume that all code you read is perfect.

Also, we use camelCase instead of underscored variable names